flowr-compilers/src/main/java/com/fueled/flowr/compilers/DeepLinkAnnotationCompiler.java

```diff
@@ -0,0 +1,306 @@
+
+import com.fueled.flowr.annotations.DeepLink;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+import com.squareup.javapoet.TypeVariableName;
+import com.squareup.javapoet.WildcardTypeName;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+
+
+/**
+ * Annotation processor that will generate the FlowrDeepLinkHandlerImpl based on the information provided by {@link com.fueled.flowr.annotations.DeepLink}.
+ */
+@SupportedAnnotationTypes("com.fueled.flowr.annotations.DeepLink")
+@SupportedSourceVersion(SourceVersion.RELEASE_7)
+public class DeepLinkAnnotationCompiler extends AbstractProcessor {
+    private static final String deeplinkFormat = "addFragment($S, (Class<? extends T>) $T.class)";
+    private static final String HANDLER_FILE_NAME = "FlowrDeepLinkHandlerImpl";
+    private static final String FLOWR_PACKAGE_NAME = "com.fueled.flowr";
+    private static final String FLOWR_INTERNAL_PACKAGE_NAME = "com.fueled.flowr.internal";
+
+    private static final TypeVariableName fragmentTypeVariableName = TypeVariableName.get("T", ClassName.get("android.support.v4.app", "Fragment"), ClassName.get(FLOWR_PACKAGE_NAME, "FlowrFragment"));
+    private static final ParameterizedTypeName classExtendsFragmentTypeName = ParameterizedTypeName.get(ClassName.get(Class.class), WildcardTypeName.subtypeOf(fragmentTypeVariableName));
+    private static final ClassName FlowrDeepLinkInfoClassName = ClassName.get(FLOWR_INTERNAL_PACKAGE_NAME, "FlowrDeepLinkInfo");
+
+    private static final ClassName bundleClassName = ClassName.get("android.os", "Bundle");
+    private static final ClassName uriClassName = ClassName.get("android.net", "Uri");
+    private static final ParameterizedTypeName listStringTypeName = ParameterizedTypeName.get(List.class, String.class);
+
+    /**
+     * Check if a list of string contains the same substring.
+     *
+     * @param strings The list of String to search
+     * @param pos     The position until where the character are the same.
+     * @return
+     */
+    private static boolean allCharactersAreSame(List<String> strings, int pos) {
+        String first = strings.get(0);
```

Comment:
@chetan-fueled no need, because if there is no annotations, the compiler is not executed. So the list of pattern is never empty.